### PR TITLE
Sales

### PR DESCRIPTION
**Pull request by okellogabrielinnocent**
- Create a SaleController
- Create a SaleRepository
- Create ItemSold event to update quantity

### CODE REVIEW

File: database/migrations/2020_03_15_152445_create_sales_table.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class CreateSalesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('sales', function (Blueprint $table) {
```
Comment (jmatembu) [inline database/migrations/2020_03_15_152445_create_sales_table.php]: @okellogabrielinnocent Aren't we missing important fields for this table? Fields such as `customer_id`, `staff_id`?

Comment (jmatembu) [inline database/migrations/2020_03_15_152445_create_sales_table.php:16]: @okellogabrielinnocent Aren't we missing important fields for this table? Fields such as customer_id, staff_id?

**Review by jmatembu — CHANGES_REQUESTED** [review]

File: database/migrations/2020_03_15_152445_create_sales_table.php (hunk 1 above)
Comment (okellogabrielinnocent) [reply database/migrations/2020_03_15_152445_create_sales_table.php:16]: Was still working on it that's why I called it `WIP`(Work In Progress)

Comment (jmatembu) [reply database/migrations/2020_03_15_152445_create_sales_table.php:16]: Oh okay, I hadn't noticed the `WIP` label

**Comment by okellogabrielinnocent** [comment]
[Sale](https://trello.com/c/PM9WwRsL/8-sale)

File: database/migrations/2020_03_15_152445_create_sales_table.php (hunk 1 above)
Comment (okellogabrielinnocent) [reply database/migrations/2020_03_15_152445_create_sales_table.php:16]: @jmatembu please take time and review

**Pull request merged**